### PR TITLE
Various tweaks and fixes, cherry picked from an obsolete branch

### PR DESCRIPTION
**Pull request by bramstoeller**

### CODE REVIEW

File: src/power_grid_model_io/filters/__init__.py [hunk 1]
```diff
@@ -58,3 +67,53 @@ def complex_inverse_imaginary_part(real: float, imag: float) -> float:
...
     """
     return (1.0 / (real + 1j * imag)).imag
+
+
+def get_winding(winding: str, neutral_grounding: bool = True) -> WindingType:
+    """
+    Return the winding type as an enum value, based on the string representation
+    """
+    winding_type = WINDING_TYPES[winding.upper()]
+    if not neutral_grounding:
```
Comment (petersalemink95) [inline src/power_grid_model_io/filters/__init__.py:77]: If it possible to have a `winding_type` with neutral like `wye_n` with `neutral_grounding = False?`

File: tests/unit/filters/test_filters.py [hunk 2]
```diff
@@ -9,8 +10,11 @@
 from power_grid_model_io.filters import (
```
Comment (petersalemink95) [inline tests/unit/filters/test_filters.py:10]: There is no unit test for `get_winding`, `all_true`, `any_true`

**Review by petersalemink95 — CHANGES_REQUESTED** [review]

**Comment by petersalemink95** [comment]
Can we put the filters in `filters/_flters.py` and then import the functions in the `__init__`?
I don't think we should define the functions within the` __init__ ` file

File: src/power_grid_model_io/filters/__init__.py (hunk 1 above)
Comment (bramstoeller) [reply src/power_grid_model_io/filters/__init__.py]: I don't think so. I guess it's rudimentary code, written when z-winding wasn't supported yet. @TonyXiang8787 Can we remove this if statement (and thereby the neutral_grounding parameter)?

File: tests/unit/filters/test_filters.py (hunk 2 above)
Comment (bramstoeller) [reply tests/unit/filters/test_filters.py:10]: Noted. I'll add them.

**Comment by bramstoeller** [comment]
> Can we put the filters in `filters/_flters.py` and then import the functions in the `__init__`? I don't think we should define the functions within the`__init__` file

Yes, done.

File: src/power_grid_model_io/filters/__init__.py (hunk 1 above)
Comment (TonyXiang8787) [reply src/power_grid_model_io/filters/__init__.py]: It is possible to have winding `wye_n` but neutral is not grounded.

So we need this logic.

Comment (bramstoeller) [reply src/power_grid_model_io/filters/__init__.py]: ok

File: tests/unit/filters/test_filters.py (hunk 2 above)
Comment (bramstoeller) [reply tests/unit/filters/test_filters.py:11]: Done.

**Pull request merged**